pkg/authn/authsvc/authn.go

```diff
@@ -47,7 +47,7 @@ func (a authentication) Authenticate(ctx context.Context, token string) (authn.S
 			return authn.Session{}, errors.Wrap(errors.ErrAuthentication, err)
 		}
 
-		return authn.Session{Type: authn.PersonalAccessToken, PatID: res.GetId(), UserID: res.GetUserId()}, nil
+		return authn.Session{Type: authn.PersonalAccessToken, PatID: res.GetId(), UserID: res.GetUserId(), Role: authn.Role(res.GetUserRole())}, nil
```

Comment:
Why don't we return the Token Type in gRPC response ? 

May be in next PR not in this PR 

Return Token type in gRPC response of AuthenticatePAT and Authenticate

Then we can Combine Authentication and AuthenticationPAT to single gRPC endpoint and combine service functions
Created a ticket for this here https://github.com/absmach/supermq/issues/3063